### DISCUSSION TITLE
Unnecessary whitespaces in command lines argument

### DISCUSSION THREAD

**Discussion by ansarigulshad** [apache/airflow#34115] [General]
We are currently using Airflow version 1.10.11. Our security operations (secops) vulnerability scan report has identified over 200+ white spaces in the output of the airflow command line process:

```
Commands: gunicorn: master [airflow-webserver]
Working directory: /opt/airflow
Username: airflow
There are 200+ whitespaces at the end.
```

It shows something like below. When checked from terminal & config file, we didn't find any white space. Please let us know if there's a way to fix this.
 
```
    "process.args": [
      "gunicorn: master [airflow-webserver]",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      "",
      ""
    ],
```

![image](https://github.com/apache/airflow/assets/5801838/9219a4b6-4390-48fc-b151-529e4f3b8fae)

**Comment by Taragolis** [comment]
> We are currently using Airflow version 1.10.11

Airflow 1.10.* is EOL for about 2 years, see [Version Life Cycle](https://github.com/apache/airflow#version-life-cycle).

> Please let us know if there's a way to fix this.

The long therm strategy should be migrate to Airflow 2.x

**Comment by ansarigulshad** [comment]
Sure, we will verify this vulnerability on 2.x version and create new thread if we find any issues.

Thank You

**Reply by Taragolis** [reply]
Please also have a look https://github.com/apache/airflow/security/policy

**Comment by potiuk** [comment]
> The long therm strategy should be migrate to Airflow 2.x

The long OVERDUE strategy should be migrate to Airflow 2.x . 

@ansarigulshad . You are using software that stopped receiving any fixes more than 2 years ago. Your company is shooting themselves in their foot by not upgrading. There were probably few 10s of vulnerabilities fixed since Airflow 1.10 reached end of life. If you care about security - you must upgrade  ASAP, otherwise your business is endangered.

**Comment by potiuk** [comment]
> Sure, we will verify this vulnerability on 2.x version and create new thread if we find any issues.

As per https://github.com/apache/airflow/security/policy - please don't report any issues unless you provide assessment of how it impacts security and reproduction scenario of exploitation.
